dm: add LOCK TABLES privilege note for managed MySQL sources (#22605)
pingcap/docs: dm/quick-start-with-dm.md

    GRANT PROCESS, BACKUP_ADMIN, RELOAD, REPLICATION SLAVE, REPLICATION CLIENT, SELECT ON *.* TO 'tidb-dm'@'%';
    ```

    > **Note:**
    >

>     > If your MySQL source is a managed MySQL service (such as Amazon RDS, Aurora, ApsaraDB RDS for MySQL, Azure Database for MySQL, or Google Cloud SQL), also grant the `LOCK TABLES` privilege. For more information, see [DM-worker privileges](/dm/dm-worker-intro.md#upstream-database-user-privileges).

Reviewer: It seems that "or" is better here, because the list gives alternative examples of a single managed MySQL service, not all of them together.

    GRANT PROCESS, BACKUP_ADMIN, RELOAD, REPLICATION SLAVE, REPLICATION CLIENT, SELECT ON *.* TO 'tidb-dm'@'%';
    ```

>     > **Note:** If your MySQL source is a managed service (such as Amazon RDS, Aurora, ApsaraDB RDS for MySQL, Azure Database for MySQL, or Google Cloud SQL), also grant `LOCK TABLES`. See [DM-worker privileges](/dm/dm-worker-intro.md#upstream-database-user-privileges) for details.

Reviewer: Suggestion to match the repository note format and keep this note aligned with the earlier one:

```suggestion
    > **Note:**
    >
    > If migrating from a managed MySQL service where `FLUSH TABLES WITH READ LOCK` is not permitted, also grant `LOCK TABLES`. For details, see [DM-worker privileges](/dm/dm-worker-intro.md#upstream-database-user-privileges).
```

    GRANT PROCESS, BACKUP_ADMIN, RELOAD, REPLICATION SLAVE, REPLICATION CLIENT, SELECT ON *.* TO 'tidb-dm'@'%';
    ```

    > **Note:**
    >

>     > If your MySQL source is a managed MySQL service (such as Amazon RDS, Aurora, ApsaraDB RDS for MySQL, Azure Database for MySQL, or Google Cloud SQL), also grant the `LOCK TABLES` privilege. For more information, see [DM-worker privileges](/dm/dm-worker-intro.md#upstream-database-user-privileges).

Reviewer: It seems that "or" is better here, because the list gives alternative examples of a single managed MySQL service, not all of them together.
